File: app/src/main/java/com/kk/app/mobile/web/register/FunctionRegisterManager.java
package com.kk.app.mobile.web.register;

import android.support.annotation.NonNull;

import com.kk.app.mobile.web.NavigationFunction;

import java.util.List;

import com.kk.app.web.function.register.IFunction;
import com.kk.app.web.function.register.IFunctionRegisterManager;

/**
 * @author kk
 * @datetime 2019-08-06
 * @desc
 */
public class FunctionRegisterManager implements IFunctionRegisterManager {

    @Override
    public void onBind(@NonNull List<IFunction> functions) {
        addFunction(functions,new NavigationFunction());
    }

    private void addFunction(@NonNull List<IFunction> functions, IFunction iFunction) {
        functions.add(iFunction);
    }
}

File: app/src/main/java/com/kk/app/mobile/web/register/FunctionRegisterManager.kt
package com.kk.app.mobile.web.register

import com.kk.app.mobile.web.NavigationFunction
import com.kk.app.web.function.register.IFunction
import com.kk.app.web.function.register.IFunctionRegisterManager

/**
 * @author kk
 * @datetime 2019-08-06
 * @desc
 */
class FunctionRegisterManager : IFunctionRegisterManager {
    override fun onBind(functions: MutableList<IFunction>) {
        addFunction(functions, NavigationFunction())
    }

    private fun addFunction(functions: MutableList<IFunction>, iFunction: IFunction) {
        functions.add(iFunction)
    }
}